code/modules/trade/datums/_trade_station.dm

```diff
@@ -0,0 +1,156 @@
+#define good_data(nam, rand_list) list("name" = nam, "amount_range" = rand_list)
+#define custom_good_name(nam) good_data(nam, null)
+#define custom_good_amount_range(rand_list) good_data(null, rand_list)
+
+/datum/trade_station
+	var/name
+	var/desc
+	var/list/icon_states = "station"
+	var/offer_type
+	var/offer_price
+	var/offer_amount
+
+	var/spawn_always = FALSE
+	var/spawn_probability = 30
+	var/spawn_cost = 1
+	var/start_discovered = FALSE
+	var/list/linked_with //trade 'stations' or 'station' that must spawn with //list or path
+
+	var/list/forced_overmap_zone //list(list(minx, maxx), list(miny, maxy))
+	var/ovemap_opacity = 0
+
+	var/list/name_pool = list()
+
+	var/markup = 0
+	var/list/assortiment = list()
+	var/list/offer_types = list()
+	
+	var/list/amounts_of_goods = list()
+
+	var/obj/effect/overmap_event/overmap_object
+	var/turf/overmap_location
+
+/datum/trade_station/New(init_on_new)
+	. = ..()
+	if(init_on_new)
+		init_src()
+
+/datum/trade_station/proc/init_src()
+	if(name)
+		crash_with("Some retard gived trade station a name before init_src, not thought name_pool. ([type])")
```

Comment:
This is not english, though spooky broken english is more effective at motivating someone if they make a mistake.